frappe/model/rename_doc.py

```diff
@@ -94,9 +94,6 @@ def rename_doc(doctype, old, new, force=False, merge=False, ignore_permissions=F
 	else:
 		new_doc.add_comment('Edit', _("renamed from {0} to {1}").format(frappe.bold(old), frappe.bold(new)))
 
-	if merge:
-		frappe.delete_doc(doctype, old)
```

Comment:
why? they still have the option to recover the deleted document `Deleted Documents`.

Reply:
When you recover that document, it is assigned a new name. If an issue with xxx1 is deleted and recovered it has a new name xxx5, also the person who is merging should decide whether to delete it or not, system should not delete it automatically.